pages/glossary/programmatic-section-of-content.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Programmatic section of content
+key: programmatic-section-of-content
+unambiguous: true
+objective: true
+---
+
+A _programmatic section of content_ is either an [explicit section of content][] or an [implicit section of content][].
+
+Programmatic sections of content may contain nested programmatic subsections of content.
+
+Programmatic sections of content may have zero or one heading [associated][] with them.
+
+**Note:** This definition roughly follows the [algorithm for creating an outline][outline algorithm], but using different elements for defining sections. The main focus is here on the [semantic role][] rather than the element itself, and all [landmark roles][landmark] are considered. Additionally, programmatic sections of content include content from [shadow trees][] and [nested browsing contexts][] which is ignored by the [algorithm for creating an outline][outline algorithm].
+
+**Note:** When elements and roles are used according to their semantics, programmatic sections of content and [sections of content][] should match perfectly.
+
+#### Assumption
```

Comment:
I'm not a fan of putting assumptions into definitions. I understand this is something we want to document is part of this assumption, but this belongs in the assumption section of the rule.